### PR TITLE
clarify hierarchy of different detector group base classes

### LINKED ISSUE #801: how do detector  base classes relate?
We can describe better in the docstring (first line of documentation) how these groups relate.  Be concise.

* NXdetector_group describes the arrangement of several NXdetector groups
* NXdetector can use one or more NXdetector _module groups

### PR DESCRIPTION
**Pull request by prjemian**
fixes #801

### CODE REVIEW

**Comment by prjemian** [comment]
If this passes travis-ci, then ready for review.

**Comment by prjemian** [comment]
More than just the docstring, the text of each of these three base classes should be consistent.

**Comment by prjemian** [comment]
It is clear there are documentation problems with NXdetector_group field `group_index`.  Those problems should be resolved in a separate issue.

**Comment by prjemian** [comment]
For consistency of wording, consider the wording on use of these groups in NXmx.

**Comment by prjemian** [comment]
NXmx provides a use case example.

**Comment by prjemian** [comment]
When clarifying, do not introduce new terms such as *detector element* when referring to a *detector*.

**Comment by prjemian** [comment]
In NXmx group DETECTOR_GROUP, 

> Each detector element is represented as an NXdetector group with its own detector data array. Each detector data array may be further decomposed into array sections by use of NXdetector_module groups. The names are given in the group names field.

change to

> Each detector is represented as an NXdetector group with its own detector data array. Each detector data array may be further decomposed into array sections by use of NXdetector_module groups. The names are given in the group names field.

under `group_names`:

> An array of the names of the detector elements or hierarchical groupings of detector elements.

changes to 

> An array of the names of the NXdetector groups or hierarchical groupings of detector elements.

**Comment by prjemian** [comment]
note: In NXmx DETECTOR_GROUP, the `group_names` documentation uses *elements* as well.

**Comment by phyy-nx** [comment]
Ready for review by @yayahjb and/or anyone else

File: applications/NXmx.nxdl.xml [hunk 1]
```diff
@@ -195,19 +195,24 @@
...
+						NXdetector_module groups.  Detectors can be grouped logically
+						together using NXdetector_group. Groups can be further grouped
+						hierarchically in a single NXdetector_group (for example, if
+						there are multiple detectors at an endstation or multiple 
+						endstations at a facility).  Alternatively, multiple 
+						NXdetector_groups can be provided.
 
 						The groups are defined hierarchically, with names given
-						in the group_names field, unique identifiing indices given
+						in the group_names field, unique identifing indices given
```
Comment (prjemian) [inline applications/NXmx.nxdl.xml:211]: identifying

File: base_classes/NXdetector_group.nxdl.xml [hunk 2]
```diff
@@ -27,37 +27,55 @@
...
-		of "tube1", "tube2" and "tube3" then group_names would be the string "bank1,
-		bank1/tube1, bank1/tube2,bank1/tube3" group_index would be {1,2,3,4} group_parent
-		would be {-1,1,1,1}
-		
-		The mapping array is interpreted as
-		group 1 is a top level group containing groups 2, 3 and 4
-		
-		A ``group_index`` array in
-		``NXdetector`` gives the base group for a detector element.
+		Optional logical grouping of detectors. When used, describes
```
Comment (prjemian) [inline base_classes/NXdetector_group.nxdl.xml:30]: MUST be on one line only.

Remove first word `Optional` per discussion this morning.  (Redundant with `When used, ` ...)

**Comment by yayahjb** [comment]
I've tried to align NXmx and NXdetector_group to the most recent versions.  I think they are ready for merge.

File: base_classes/NXdetector_group.nxdl.xml [hunk 3]
```diff
@@ -57,25 +57,34 @@
...
-		hierarchical groupings of detectors.</doc>
+	 	<doc>
+			An array of the names of the detectors given in NXdetector
+			groups or the names of hierarchical groupings of detectors
+			given as names of NXdetector_group groups or in
+			NXdetector_group group_names and group_parent fields as
+			having children.
+			
+			Specified in the base classes as comma separated list of names,
+			but new code should use an array of names as quoted strings.
```
Comment (phyy-nx) [inline base_classes/NXdetector_group.nxdl.xml:67]: I think we can drop these two lines from NXdetector_group and NXmx?

File: applications/NXmx.nxdl.xml [hunk 4]
```diff
@@ -195,19 +195,24 @@
 	
 				<group type="NXdetector_group" recommended="true">
 					<doc>
-						Optional logical grouping of detector elements.
+						Optional logical grouping of detectors.
 
-						Each detector element is represented as an NXdetector group
+						Each detector is represented as a NXdetector
```
Comment (PeterC-DLS) [inline applications/NXmx.nxdl.xml:200]: an NXdetector

**Comment by prjemian** [comment]
LGTM

File: applications/NXmx.nxdl.xml [hunk 5]
```diff
@@ -222,41 +227,38 @@
...
+							An array of the names of the detectors or the names of
+							hierarchical groupings of detectors.
 						</doc>
 					</field>
 
 					<field name="group_index" type="NX_INT">
 						<doc>
-							An array of unique indices for detector elements or groupings
-							of detector elements.
+							An array of unique indices for detectors or groupings
```
Comment (PeterC-DLS) [inline applications/NXmx.nxdl.xml:237]: unique identifiers rather than indices?

**Comment by benajamin** [comment]
This PR is only fixing a bug and clarifying documentation and so doesn't require a NIAC decision.

**Review by phyy-nx — APPROVED** [review]

**Review by yayahjb — APPROVED** [review]
Thank God

**Pull request merged**